src/erpbrasil/base/fiscal/recopi.py

```
@@ -0,0 +1,162 @@
+# coding=utf-8
+# Copyright (C) 2023  Daniel Venancio - KMEE
+# License MIT - See https://opensource.org/license/mit
+import datetime as dt
+
+CURRENT_YEAR = dt.date.today().year
+CURRENT_MONTH = dt.date.today().month
+CURRENT_DAY = dt.date.today().day
+VERIFICATION_DIGIT_WEIGHT = [
+    19,
+    18,
+    17,
+    16,
+    15,
+    14,
+    13,
+    12,
+    11,
+    10,
+    9,
+    8,
+    7,
+    6,
+    5,
+    4,
+    3,
+    2,
+    1,
+]
+
+
+def is_valid_recopi(recopi):
+    if _check_recopi_size(recopi):
+        if _check_recopi_format(recopi):
+            return True
+    return False
+
+
+def generate_valid_recopi():
+    recopi_date = _generate_recopi_date()
+    recopi_hour = _generate_recopi_hour()
+    recopi_four_digits = _generate_recopi_four_digits()
+    recopi_verification_digits = _generate_recopi_verification_digits(
+        recopi_date + recopi_hour + recopi_four_digits
+    )
+    recopi = recopi_date + recopi_hour + recopi_four_digits + recopi_verification_digits
+    return recopi
+
+
+def _check_recopi_size(recopi):
+    if len(recopi) == 20:
+        return True
+    return False
+
+
+def _check_recopi_format(recopi):
+    return (
+        _check_recopi_date(recopi)
+        and _check_recopi_time(recopi)
+        and _check_recopi_verification_digits(recopi)
+    )
+
+
+def _check_recopi_date(recopi):
+    recopi_year, recopi_month, recopi_day = get_recopi_date(recopi)
+    try:
+        dt.date(
+            year=recopi_year, month=recopi_month, day=recopi_day
+        )  # _check if it's a calendar date
+        if recopi_year > CURRENT_YEAR:
+            return True
+        elif recopi_year == CURRENT_YEAR and recopi_month > CURRENT_MONTH:
+            return True
+        elif recopi_year < 2013:
+            return True
+
+        return False
+    except:
```

Corrigir o uso do except
Não utilize o bare except, pois ele captura qualquer exceção.
Exemplo: except Exception

https://peps.python.org/pep-0008/#programming-recommendations